### PR TITLE
[IA-2516] Add TraceId to ErrorReport

### PR DESCRIPTION
**Pull request by ehanna4**
Add TraceId to ErrorReport so we can show traceid in a more user friendly way on Leo

No version bump - adding an optional parameter

**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [x] Get two thumbsworth of PR review
- [x] Verify all tests go green (CI _and_ coverage tests)
- [x] Squash commits and **merge to develop**
- [x] Delete branch after merge

### CODE REVIEW

**Review by Qi77Qi — APPROVED** [review]
first wb-libs PR! 🎉

**Review by Qi77Qi — CHANGES_REQUESTED** [review]
https://broadinstitute.slack.com/archives/GB1VAM14J/p1613144892047500?thread_ts=1613071326.027800&cid=GB1VAM14J

File: README.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ Workbench utility libraries, built for Scala 2.12 and 2.13. You can find the ful
 
 Contains utility functions for publishing custom metrics using openTelemetry (openCensus and openTracing). 
 
-Latest SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-error-reporting" % "0.1-92fcd96"`
```
Comment (Qi77Qi) [inline README.md:99]: need to update this for `model` too :)

**Review by Qi77Qi — APPROVED** [review]
:shipit:

File: model/CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ This file documents changes to the `workbench-model` library, including notes on
 
 ## 0.14
 
-SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-model" % "0.14-2e155f0"`
+SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-model" % "0.14-TRAVIS-REPLACE-ME"`
```
Comment (Qi77Qi) [inline model/CHANGELOG.md:7]: oh actually, can you add a comment, something like "add `TraceId` to `ErrorReport`"

File: model/src/main/scala/org/broadinstitute/dsde/workbench/model/ErrorReport.scala [hunk 3]
```diff
@@ -28,57 +29,60 @@ case class ErrorReportSource(source: String)
...
   def apply(message: String,
             statusCode: Option[StatusCode],
             causes: Seq[ErrorReport],
             stackTrace: Seq[StackTraceElement],
-            exceptionClass: Option[Class[_]]
+            exceptionClass: Option[Class[_]],
+            traceId: Option[TraceId]
   )(implicit source: ErrorReportSource): ErrorReport =
-    ErrorReport(source.source, message, statusCode, causes, stackTrace, exceptionClass)
+    ErrorReport(source.source, message, statusCode, causes, stackTrace, exceptionClass, traceId)
```
Comment (kyuksel) [inline model/src/main/scala/org/broadinstitute/dsde/workbench/model/ErrorReport.scala:85]: Adding a parameter to a class constructor (`def apply()` in this case) with no default value seems like a breaking change to me as existing callers of the constructor won't be able to keep calling it the same way. Am I missing something?

**Review by kyuksel — CHANGES_REQUESTED** [review]
Congratulations on your first workbench-libs PR, Emily 👏🏼 

Left a comment regarding the nature of the change (i.e. breaking) as opposed to what's implied from the change log (i.e. non-breaking).

**Review by kyuksel — APPROVED** [review]
:shipit:

**Pull request merged**